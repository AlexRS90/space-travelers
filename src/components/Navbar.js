import React from 'react';
import { BrowserRouter as Router, NavLink } from 'react-router-dom';
import logo from '../images/logo.png';
import './navbar.css';

const Navbar = () => (
  <Router>
    <nav className="navbar navbar-light bg-light p-3" data-testid="nav">
      <div className="container">
        <a className="navbar-brand" href="/">
          <img src={logo} alt="logo" className="me-3" width="30" height="24" />
          Space Travelers Hub
        </a>
        <ul className="nav-menu">
          <li><NavLink to="/rockets" activeclassname="active">Rockets</NavLink></li>
          <li><NavLink to="/missions" activeclassname="active">Missions</NavLink></li>
          <li><NavLink to="/myProfile" activeclassname="active">My Profile</NavLink></li>
        </ul>
      </div>
    </nav>
  </Router>

);

export default Navbar;
