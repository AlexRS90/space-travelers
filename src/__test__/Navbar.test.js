import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

afterEach(() => {
  cleanup();
});

describe('render navbar to the screen', () => {
  test('render navbar to the screen', () => {
    render(<Router><Navbar /></Router>);
    const navbarElement = screen.getByTestId('nav');
    expect(navbarElement).toBeInTheDocument();
  });

  test('matches the snapshot', () => {
    const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
