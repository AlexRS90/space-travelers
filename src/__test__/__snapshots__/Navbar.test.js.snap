// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render navbar to the screen matches the snapshot 1`] = `
<nav
  className="navbar navbar-light bg-light p-3"
  data-testid="nav"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="logo"
        className="me-3"
        height="24"
        src="logo.png"
        width="30"
      />
      Space Travelers Hub
    </a>
    <ul
      className="nav-menu"
    >
      <li>
        <a
          activeclassname="active"
          className=""
          href="/rockets"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          className=""
          href="/myProfile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
